doctr/models/recognition/master/pytorch.py

```diff
@@ -239,21 +272,33 @@ def forward(
 
         if target is not None:
             # Compute target: tensor of gts and sequence lengths
-            gt, _ = self.compute_target(target)
-            tgt_mask = self.make_mask(torch.from_numpy(gt))
+            gt, seq_len = self.compute_target(target)
+            gt, seq_len = torch.from_numpy(gt).to(dtype=torch.long), torch.tensor(seq_len)
+
+        if self.training:
+            if target is None:
+                raise AssertionError("In training mode, you need to pass a value to 'target'")
+            tgt_mask = self.make_mask(gt)
             # Compute logits
-            output = self.decoder(torch.from_numpy(gt), encoded, tgt_mask, None)
+            output = self.decoder(gt, encoded, tgt_mask, None)
             logits = self.linear(output)
 
         else:
             _, logits = self.decode(encoded)
```

Comment:
If we don't need the `ys`, perhaps we should avoid computing them in `self.decode`?

Reply:
we need to iterate over ys in decode, it is a kind of buffer